core/domain/user_services.py

```diff
@@ -656,3 +656,36 @@ def get_user_dashboard_stats(user_id):
             'total_plays': 0,
             'average_ratings': None
         }
+
+def get_weekly_dashboard_stats(user_id):
+    """Returns list which contains the dashboard stats of a user"""
+
+    model = user_models.UserStatsModel.get(user_id, strict=False)
+
+    if model and model.weekly_creator_stats_list:
+        return model.weekly_creator_stats_list
+    else:
+        return None
+
+def save_last_dashboard_stats(user_id):
```

Comment:
Actually, I think it makes more sense to create a mapreduce job to do this instead. (Do a one-off job, not a continuous job, and let the cron start the job). It might not be possible to actually load all the users to memory.